Sources/Misc/Codable/EnsureNonEmptyArrayDecodable.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  Copyright RevenueCat Inc. All Rights Reserved.
+//
+//  Licensed under the MIT License (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      https://opensource.org/licenses/MIT
+//
+//  EnsureNonEmptyArrayDecodable.swift
+//
+//  Created by Nacho Soto on 7/17/23.
+
+import Foundation
+
+/// A property wrapper that ensures decoded arrays aren't empty.
+/// - Example:
+/// ```
+/// struct Data {
+///     @EnsureNonEmptyArrayDecodable var values: [String] // fails to decode if array is empty
+/// }
+/// ```
+@propertyWrapper
+struct EnsureNonEmptyArrayDecodable<Value: Codable> {
+
+    struct Error: Swift.Error {}
+
+    var wrappedValue: [Value]
+
+}
+
+extension EnsureNonEmptyArrayDecodable: Equatable where Value: Equatable {}
+extension EnsureNonEmptyArrayDecodable: Hashable where Value: Hashable {}
+
+extension EnsureNonEmptyArrayDecodable: Decodable {
+
+    init(from decoder: Decoder) throws {
+        let container = try decoder.singleValueContainer()
+        let array = try container.decode([Value].self)
+
+        if array.isEmpty {
+            throw Error()
```

Comment:
This seems a bit of a generic error... Would it be possible to include the property that is causing this error?

Reply:
It’s `EnsureNonEmptyArray.Error`. When you decode the top level object `JSONDecoder` will include the information for what key failed :)